io/disk/tiobench.py

```diff
@@ -128,9 +126,174 @@ def create_fs(self, l_disk, mountpoint, fstype):
             self.fail("Mounting disk %s on directory %s failed"
                       % (l_disk, mountpoint))
 
+    def pre_cleanup(self):
+        """
+        cleanup the disk before test starts on it
+        """
+        self.log.info("checking if the disk is mounted....")
+        if self.is_disk_mounted(self.disk):
+            self.log.info("disk mounted ....unmounting it.. calling delete_fs")
+            self.delete_fs(self.disk)
+        elif self.is_dir_mounted(self.dir):
+            self.log.info("%s mounted.... unmounting it" % self.dir)
+            self.delete_fs(self.disk)
+        elif self.is_fs_exists(self.disk):
+            self.log.info("found fs.... deleting it..")
+            self.delete_fs(self.disk)
+        self.log.info("checking ...lv/vg existance...")
+        if lv_utils.lv_check(self.vgname, self.lvname):
+            self.log.info("found lv existance... deleting it")
+            self.delete_lv()
+        elif lv_utils.vg_check(self.vgname):
+            self.log.info("found vg existance ... deleting it")
+            lv_utils.vg_remove(self.vgname)
+        self.log.info("checking for sraid existance...")
+        if self.is_raid_exists():
+            self.log.info("found sraid existance... deleting it")
+            self.delete_raid()
+        self.log.info("\n End of pre_cleanup")
+
+    def delete_raid(self):
+        '''
+        it checks for existing of raid and deletes it
+        '''
+        self.log.info("deleting Sraid %s" % self.raid_name)
+
+        def is_raid_deleted():
+            self.sraid.stop()
+            self.sraid.clear_superblock()
+            self.log.info("checking for raid metadata")
+            cmd = "wipefs -af %s" % self.disk
+            process.system(cmd, shell=True, ignore_status=True)
+            if self.is_raid_exists():
+                return False
+            return True
+        self.log.info("checking lvm_metadata on %s" % self.raid_name)
+        cmd = 'blkid -o value -s TYPE %s' % self.raid_name
+        out = process.system_output(cmd, shell=True,
+                                    ignore_status=True).decode("utf-8")
+        if out == 'LVM2_member':
+            cmd = "wipefs -af %s" % self.raid_name
+            process.system(cmd, shell=True, ignore_status=True)
+        if wait.wait_for(is_raid_deleted, timeout=10):
+            self.log.info("software raid  %s deleted" % self.raid_name)
+        else:
+            self.log.info("failed to delete softwareraid %s" % self.raid_name)
+
+    def delete_lv(self):
+        """
+        checks if lv/vg exists and delete them along with its metadata
+        if exists.
+        """
+        def is_lv_deleted():
+            lv_utils.lv_remove(self.vgname, self.lvname)
+            time.sleep(5)
```

Comment:
can we have other way to confirm lv device removed . instead of sleep ?

Reply:
this is already in wait_for() method. and we need a few seconds of wait time b/w vg and lv deletes.